File: API/apollo_api_services/src/main/java/com/apollo/entity/Variant.java
package com.apollo.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.*;

import javax.persistence.*;
import java.util.List;
import java.util.Set;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
@Table(name = "variant")
public class Variant {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String skuCode;
    private int stockQuantity;
    private double weight;
    private double price;
    private double salePrice;
    private String img;
    private Boolean isDeleted;

    @ManyToOne
    @JoinColumn(name = "product_id")
    @JsonBackReference
    private Product product;

    @OneToMany(mappedBy = "variant")
    @JsonIgnore
    private List<Image> images;

    @OneToMany(mappedBy = "variant")
    @JsonIgnore
    private List<Video> videos;

    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(name = "variant_optionvalue",
            joinColumns = @JoinColumn(name = "variant_id"),
            inverseJoinColumns = @JoinColumn(name = "optionvalue_id"))
    @JsonIgnore
    private List<OptionValue> optionValues;

    @OneToOne(mappedBy = "variant")
    private CartLine cartLine;

    @OneToOne(mappedBy = "variant")
    private SaveForLater saveForLater;

    @OneToMany(mappedBy = "variant")
    @JsonIgnore
    private List<Review> reviews;

    @OneToMany(mappedBy = "variant")
    @JsonIgnore
    private Set<ShopOrder> shopOrders;
}

File: Mobile App/Apollo Deliver/lib/Models/Variant.dart
import 'package:apollodeliver/Models/CartLine.dart';
import 'package:apollodeliver/Models/Image.dart';
import 'package:apollodeliver/Models/OptionValue.dart';
import 'package:apollodeliver/Models/Product.dart';
import 'package:apollodeliver/Models/Review.dart';
import 'package:apollodeliver/Models/SaveForLater.dart';
import 'package:apollodeliver/Models/ShopOrder.dart';
import 'package:apollodeliver/Models/Video.dart';

class Variant {
  int? id;
  String? name;
  String? skuCode;
  int? stockQuantity;
  double? weight;
  double? price;
  double? salePrice;
  String? img;
  bool? isDeleted;
  Product? product;
  List<Image>? images;
  List<Video>? videos;
  List<OptionValue>? optionValues;
  CartLine? cartLine;
  SaveForLater? saveForLater;
  List<Review>? reviews;
  Set<ShopOrder>? shopOrders;

  Variant({
    this.id,
    this.name,
    this.skuCode,
    this.stockQuantity,
    this.weight,
    this.price,
    this.salePrice,
    this.img,
    this.isDeleted,
    this.product,
    this.images,
    this.videos,
    this.optionValues,
    this.cartLine,
    this.saveForLater,
    this.reviews,
    this.shopOrders,
  });

  factory Variant.fromJson(Map<String, dynamic> json) {
    return Variant(
      id: json['id'],
      name: json['name'],
      skuCode: json['skuCode'],
      stockQuantity: json['stockQuantity'],
      weight: (json['weight'] as num?)?.toDouble(),
      price: (json['price'] as num?)?.toDouble(),
      salePrice: (json['salePrice'] as num?)?.toDouble(),
      img: json['img'],
      isDeleted: json['isDeleted'],
      product: json['product'] != null ? Product.fromJson(json['product']) : null,
      images: (json['images'] as List?)?.map((item) => Image.fromJson(item)).toList(),
      videos: (json['videos'] as List?)?.map((item) => Video.fromJson(item)).toList(),
      optionValues: (json['optionValues'] as List?)?.map((item) => OptionValue.fromJson(item)).toList(),
      cartLine: json['cartLine'] != null ? CartLine.fromJson(json['cartLine']) : null,
      saveForLater: json['saveForLater'] != null ? SaveForLater.fromJson(json['saveForLater']) : null,
      reviews: (json['reviews'] as List?)?.map((item) => Review.fromJson(item)).toList(),
      shopOrders: (json['shopOrders'] as List?)?.map((item) => ShopOrder.fromJson(item)).toSet(),
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'name': name,
    'skuCode': skuCode,
    'stockQuantity': stockQuantity,
    'weight': weight,
    'price': price,
    'salePrice': salePrice,
    'img': img,
    'isDeleted': isDeleted,
    'product': product?.toJson(),
    'images': images?.map((item) => item.toJson()).toList(),
    'videos': videos?.map((item) => item.toJson()).toList(),
    'optionValues': optionValues?.map((item) => item.toJson()).toList(),
    'cartLine': cartLine?.toJson(),
    'saveForLater': saveForLater?.toJson(),
    'reviews': reviews?.map((item) => item.toJson()).toList(),
    'shopOrders': shopOrders?.map((item) => item.toJson()).toList(),
  };
}